pkg/workceptor/workceptor.go

```diff
@@ -678,25 +688,91 @@ func (w *Workceptor) unitStatusForCFR(unitID string) (map[string]interface{}, er
 	return retMap, nil
 }
 
+func (w *Workceptor) attemptRemoteAction(remoteNodeID, remoteUnitID, remoteAction string, unit *workUnit) error {
+	timeout := time.After(300 * time.Second)
+	retryInterval := 500 * time.Millisecond
+	shouldTimeout := func () bool {
+		select {
+		case <-timeout:
+			logger.Error("%s of remote work unit timed out", remoteAction)
+			return true
+		default:
+			time.Sleep(retryInterval)
+			return false
+		}
+	}
+
+	for {
+		conn, _, err := w.connectToRemote(remoteNodeID)
+		if err != nil {
+			logger.Error("Connection failed to %s: %s\n", remoteNodeID, err)
+		}
+
+		if conn != nil {
+				_, err = conn.Write([]byte(fmt.Sprintf("work %s %s\n", remoteAction, remoteUnitID)))
+			if err != nil {
+				logger.Error("Write error sending to %s: %s\n", remoteNodeID, err)
+			} else {
+				conn.Close()
+				break
+			}
+			conn.Close()
+		}
+
+		if shouldTimeout() {
+			return nil
+		}
+	}
+
+   if remoteAction == "cancel" {
```

Comment:
`attemptRemoteAction` is used both for release and cancel, in the case of cancel we wait around to see if the status actually changes.